File: Backend/Management.DL/Models/Category.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Management.DL.Models
{
    public class Category
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public Guid? ParentCategoryId { get; set; }
        public Category? ParentCategory { get; set; }
        public ICollection<Category>? SubCategories { get; set; }
        public ICollection<Product>? Products { get; set; }
        public CategoryStatus Status { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
    }
}

File: Frontend/Managment-FE/src/app/pages/models/category.ts
export interface Category {
  id: string;
  name: string;
  description?: string;
  parentCategoryId?: string | null;
  status: CategoryStatus | null; 
  createdDate: string;
  updatedDate: string;
  isEditable?: boolean; 
}

export interface PaginatedCategory {
  totalCount: number; 
  pageNumber: number; 
  pageSize: number; 
  data: Category[]; 
}

export enum CategoryStatus {
  Active = 0,
  Inactive = 1
}
